packages/backend/src/services/test-run.ts

```diff
@@ -76,7 +76,11 @@ const testRun = async (options: TestRunOptions) => {
         testRun: true,
       })
 
-    if (actionExecutionStep.isFailed || actionStep.id === untilStep.id) {
+    if (actionExecutionStep.isFailed) {
+      return { executionStep: actionExecutionStep }
+    }
+
+    if (actionStep.id === untilStep.id) {
```

Comment:
why?

Reply:
previously, i updated the execution status for test run as well, will change it back to the original code